src/components/PriceField/PriceField.tsx

```diff
@@ -56,74 +53,24 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
     hint = "",
     currencySymbol,
     name,
-    onChange,
+    onChange: onChangeBase,
     required,
     value,
     InputProps,
     inputProps
   } = props;
 
   const classes = useStyles(props);
-  const minValue = 0;
-
-  const maxDecimalLength = useMemo(
-    () => getCurrencyDecimalPoints(currencySymbol),
-    [currencySymbol]
+  const { onChange, onKeyDown, minValue, step } = usePriceField(
+    currencySymbol,
+    onChangeBase
   );
 
-  const handleChange: FormChange = e => {
-    let value = e.target.value;
-    const splitCharacter = findPriceSeparator(value);
-    const [integerPart, decimalPart] = value.split(splitCharacter);
-
-    if (maxDecimalLength === 0 && decimalPart) {
-      // this shouldn't happen - decimal character should be ignored
-      value = integerPart;
-    }
-
-    if (decimalPart?.length > maxDecimalLength) {
-      const shortenedDecimalPart = decimalPart.slice(0, maxDecimalLength);
-      value = `${integerPart}${splitCharacter}${shortenedDecimalPart}`;
-    }
-
-    onChange({
-      target: {
-        name: e.target.name,
-        value
-      }
-    });
-  };
-
-  const handleKeyPress: TextFieldProps["onKeyDown"] = e => {
-    // disallow entering e (exponent)
-    if (e.key === "e" || e.key === "E") {
-      e.preventDefault();
-    }
-    // ignore separator input when currency doesn't support decimal values
-    if (
-      maxDecimalLength === 0 &&
-      SEPARATOR_CHARACTERS.some(separator => e.key === separator)
-    ) {
-      e.preventDefault();
-    }
-  };
-
   return (
     <TextField
       className={className}
-      error={error || value < minValue}
-      helperText={
-        hint ? (
-          hint
-        ) : value < minValue ? (
-          <FormattedMessage
-            id="WHkx+F"
-            defaultMessage="Price cannot be lower than 0"
-          />
-        ) : (
-          ""
-        )
-      }
```

Comment:
Why isn't this `minValue` validation inside the new `usePriceField` hook? Are we supposed to do the `minValue` check wherever we use the `PriceField` component from now on?